src/arch/xtensa/boot_loader.c

```diff
@@ -231,6 +231,32 @@ static int32_t hp_sram_init(void)
 }
 #endif
 
+#if defined(CONFIG_ICELAKE)
+static int32_t lp_sram_init(void)
+{
+	int status;
+	int delay_count = 256;
+
+	shim_write(SHIM_LDOCTL, SHIM_LDOCTL_HPSRAM_LDO_ON);
+
+	/* query the power status of first part of LP memory */
+	/* to check whether it has been powered up. A few    */
+	/* cycles are needed for it to be powered up         */
+	status = io_reg_read(LSPGISTS);
+	while (status) {
+		idelay(delay_count);
+		status = io_reg_read(LSPGISTS);
```

Comment:
Yes, no doubt, I have added a simplified one however I think we should have some more advanced component for that. It could work base on one of the timer, offer interfaces to set timeout in both ms and ticks so other modules can import it and use as needed.